src/main/java/org/jenkinsci/plugins/badge/PublicBadgeAction.java

```diff
@@ -127,11 +128,11 @@ public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @Que
         Job<?, ?> project = getProject(job);
         Integer codeCoverage = null;
 
-        if (project.getLastSuccessfulBuild() != null) {
```

Comment:
So the problem is if you get the lastCompleted build, if it was aborted/failed then the code coverage would show it as 0 (most likely the case if a CI job fails).